__tests__/should-close-popup-on-unmount.js

```diff
@@ -1,27 +1,34 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
-import OauthPopup from "../src/index.js";
+import renderer, { act } from 'react-test-renderer';
```

Comment:
[minor] Look into [React Testing Library](https://testing-library.com/docs/react-testing-library/intro), it's quite good for basic usage and it does a lot of the `act` shenanigans for you.